src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -278,4 +314,14 @@ private function addServiceToAmbiguousType($id, $type)
         }
         $this->ambiguousServiceTypes[$type][] = $id;
     }
+
+    static private function getResourceMetadataForMethod(\ReflectionMethod $method)
+    {
+        $methodArgumentsMetadata = array();
+        foreach ($method->getParameters() as $parameter) {
+            $methodArgumentsMetadata[] = (string) $parameter;
```

Comment:
IIRC, the argument name doesn't matter (it can be changed safely), only the type hint and the order is important.